### ISSUE TITLE
Lithuanian localisation

### ISSUE THREAD

**Issue by maras** [plk/biblatex#972]
Hello.

First of all, thank you for your hard work.

I prepared Lithuanian localisation file, it can be found at: https://github.com/maras/lithuanian.lbx/blob/master/lithuanian.lbx It is based on older unofficial variant which I had to rewrite significantly.
langserbian and fromserbian are commented out because I can't compile file 03-localization-keys.tex with them: 
> Package keyval Error: langserbian undefined. 

> Package keyval Error: fromserbian undefined.

I inserted and commented out langlithuanian.

There is need to write the word "language" next to the name of a language that is why next to the language name is word "kalba" or abbreviation "k." in the language strings.
There is declension in Lithuanian, so in some cases there is need to use genitivus (translated from). 
This is especially important working with month names, so I created local forms of month in genitivus (they are commented out now).

## Answers to the localisation questions.

> What kind of dash do you use in number ranges like "35-50"?

 `\textendash` There are no any spaces around the dash.

> What kind of dash do you use in date ranges like "1st January-3rd February 2009"?

`\textendash` and there *are* spaces around the dash.

> In lists and enumerations, do you add a comma in addition to connectors like "and"? 

No, there is no comma before last "and".

>    How do you format short ordinals? Are ordinals gender/number sensitive? 

All ordinals are gender sensitive and number insensitive:
1-as 2-as 3-as... (masc.), 1-a 2-a (fem.)

> What's the gender of the word "edition"?

Masculine.

> What's the gender of the word "series" (of a journal)?

Feminine.

> What's the long date format?

[year] m. [genitivus of month name] [day] d.
2020 m. kovo 4 d.

> What happens if the day is missing from the date?

[year] m. [nominativus of month name]
2020 m. kovas

> What happens if both day and month are missing?

[year] m.
2020 m.

> Does this day format use leading zeros?

No.

> Please be specific about the spaces to be used (non-breakable, thin spaces).

I don't know about specific requirements concerning spaces in the long dates.

> What happens for date ranges with compact (1.--6. February 2010, 1. February--6. March 2010) and non-compact ranges (1. February 2010--6. February 2010)?

- Compact (they are quite rare): 2020 m. kovo 13--14 d. (endash between days without spaces, month – genitive) 
  But when there are no days: 
  2020 m. sausis -- kovas or
  1920 m. sausis -- 1922 m. kovas
  (endash with spaces between month, month – nominative)
  The same when only year: 1920 m. -- 1922 m.
- Non compact: endash with spaces between dates: 
  2019 m. gruodžio 14 d. -- 2020 m. kovo 14 d.

> What's the short date format?

yyyy-mm-dd (hyphens between date parts)
2020-03-04

> What happens if the day is missing from the date?

yyyy-mm
2020-03

> What happens if both day and month are missing?

yyyy (only date number)
2020

>  Does this day format use leading zeros?

Yes. And month too.

> Please be specific about the spaces to be used (non-breakable, thin spaces).

There are no spaces in this date format.

> What happens for date ranges with compact (1.--6. February 2010, 1. February--6. March 2010) and non-compact ranges (1. February 2010--6. February 2010)?

Endash without spaces
2020-03-04--14 (between days)
2020-03-04--2020-04-14 (between dates
2020-03--04 (between month)

> After which punctuation marks do you capitalize a word? 

After ". : ! ?" (after ":" only in bibliographies and citations)

> What's the 12h time format?/What's the 24h time format?
- 12h format: 
  [hour] val [minutes] min.
  3 val. 15 min. or
  3.15 val. or
  3.15
- 24h format:
  15 val. 15 min. or
  15.15 val. or
  15.15

Endash without spaces between hours.

>    What's the format for dates with seasons only, such as 'Summer 2010'?

[year][season nominative]
2020 pavasaris

**Comment by moewew** [comment]
Thank you very much for your contribution. I will have a closer look at what you wrote tomorrow. 

The errors with `langserbian` and `fromserbian` are expected if you use a release version of `biblatex`. Since the strings were only added very recently, they are only known to `dev`-branch `biblatex`.

The dates look like fun! Do you happen to know if Lithuanian dates are structurally similar to Latvian (https://github.com/plk/biblatex/blob/dev/tex/latex/biblatex/lbx/latvian.lbx, cf. also https://github.com/plk/biblatex/pull/708, https://github.com/plk/biblatex/pull/710) or Hungarian (https://github.com/plk/biblatex/blob/dev/tex/latex/biblatex/lbx/magyar.lbx, cf. https://github.com/plk/biblatex/issues/717) dates? That would make our lives significantly easier.

Do you want to be listed in the acknowledgement section of the documentation as a language contributor in https://github.com/plk/biblatex/blob/33468f12a4130ef288ea4fd63aaef56078e6a65d/doc/latex/biblatex/biblatex.tex#L202-L228 If so, please let me know the name you would like to be listed with.

**Comment by maras** [comment]
The dates look like fun! 



That's for shure. I didn't know some rules myself till preparation of this file.


Do you happen to know if Lithuanian dates are structurally similar to Latvian (https://github.com/plk/biblatex/blob/dev/tex/latex/biblatex/lbx/latvian.lbx, cf. also #708, #710) or Hungarian (https://github.com/plk/biblatex/blob/dev/tex/latex/biblatex/lbx/magyar.lbx, cf. #717) dates? That would make our lives significantly easier.




I've checked descriptions and Hungarian long date format from [there](https://github.com/plk/biblatex/issues/717#issuecomment-369902123) is almost exact like in Lithuanian. Hungarian: "[year]. [month name] [day]." Lithuanian: "[year] m. [month name] [day without leading zeros] d." Differences are: "[space]m." after year (not dot), "[space]d." after day (not dot) and genitivus form of a month. Dashes and spaces around them are the same.

There is one more rule: when long date is without a day, there can be not "[year] m. [month nominativus]" but "[year] m. [month genitivus] mėn." But we can omit that rule for bibliographies, I think. It is much more convenient to write only "[year] m. [month nominativus]" – other way would be too complicated and rather rare use cases (mostly for some archival documents).

Short form is like in international ISO standard, I think: [yyyy]-[mm]-[dd]. It resembles hungarian short date format too. Hungariant: "[year]. [month with leading zeros]. [day with leading zeros]." Lithuanian: "[year]-[month with leading zeros]-[day with leading zeros]" – (hyphens instead of ".[space]" and no dot after day.

One more thing, short ordinals are only numbers in the most cases, except edition (maybe I forgot something more).

Do you want to be listed in the acknowledgement section of the documentation as a language contributor in https://github.com/plk/biblatex/blob/33468f12a4130ef288ea4fd63aaef56078e6a65d/doc/latex/biblatex/biblatex.tex#L202-L228 If so, please let me know the name you would like to be listed with.



You can list my name: Valdemaras Klumbys

**Comment by moewew** [comment]
I take it the "m." and "d." that appear in the dates are the Lithuanian words for year and day? Can you confirm that? Do we need to differentiate (in theory) between "year"/"day" in nominative or "year"/"day" in genitive?

I can try to look into 

> [year] m. [month genitivus] mėn.

if that is "more correct" than "[year] m. [month nominativus]". It is probably not much more complicated than the rest we have to do for the dates. I just need your to tell me what to do here.

It is dangerous to have empty translations as in
```latex
  am               = {{}{}},
  pm               = {{}{}},
```
if there is no equivalent to "a.m." and "p.m." in Lithuanian it is better to leave the strings untranslated *and* commented out.

What do you mean by

> One more thing, short ordinals are only numbers in the most cases, except edition (maybe I forgot something more).

By default `biblatex` doesn't use ordinals in a lot of places anyway. They are basically only used for editions (`\mkbibordedition`), series (`\mkbibordseries`) and for some languages in date formats (not applicable to Lithuanian). Hungarian and to some extent Latvian are more keen on ordinals, so they use them in more places.

How would you say 

> 5th edition

> Volume 6

> Chapter 3

> pp. 245-260

> Version 2

in Lithuanian?

**Comment by moewew** [comment]
Ah, I forgot: For full support, you may also want to submit a request to `csquotes` to include Lithuanian quotation marks. See for example https://github.com/josephwright/csquotes/issues/22, https://github.com/josephwright/csquotes/issues/19.

**Comment by moewew** [comment]
Tow more things.

----

What about long date compact ranges of the form

> 1. February--6. March 2010

---

If I understand correctly, the nominative of the month name is only used in forms like

> [year] m. [nominativus of month name]

Now if

> [year] m. [month genitivus] mėn.

is more correct *and* means that we never need the nominative case, that might make things a tiny bit easier.

**Comment by moewew** [comment]
So far I got
```latex
\documentclass[lithuanian]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=authoryear, backend=biber, alldates=comp, labeldate=year]{biblatex}

\begin{filecontents}{\jobname.bib}
@book{appleby,
  author  = {Humphrey Appleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-02-01/2020-02-06},
}
@book{bppleby,
  author  = {Humphrey Bppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-02-01/2020-03-06},
}
@book{cppleby,
  author  = {Humphrey Cppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2019-03-05/2020-01-07},
}
@book{dppleby,
  author  = {Humphrey Dppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2019-07/2019-09},
}
\end{filecontents}
\addbibresource{\jobname.bib}


\begin{document}
\cite{appleby,bppleby,cppleby,dppleby}
\printbibliography
\end{document}
```
to produce
> Appleby, Humphrey (2020 m. vas. 1–6 d.). On the Importance of the Civil Service.
> Bppleby, Humphrey (2020 m. vas. 1 d. – kov. 6 d.). On the Importance of the Civil Service.
> Cppleby, Humphrey (2019 m. kov. 5 d. – 2020 m. saus. 7 d.). On the Importance of the Civil Service.
> Dppleby, Humphrey (2019 m. liep. m˙ en. – rugs. m˙ en.). On the Importance of the Civil Service.

![Screenshot of the output shown above](https://user-images.githubusercontent.com/6755835/76679463-f8d4d980-65e0-11ea-91d7-dbf66e277db8.png)

Does that look OK so far? (Modulo the questions I asked above.)

**Comment by moewew** [comment]
Is the difference between short and long form "U.S." vs "JAV" in
```latex
  patentus         = {{U.S\adddotspace patentas}{JAV\addspace pat\adddot}},
```
and
```latex
  patrequs         = {{U.S\adddotspace patento prašymas}{JAV\addspace pat\adddot\ praš\adddot}},
```
intentional?

---

I also found an inconsistency in the placement of `\lbx@lfromlang`/`\lbx@sfromlang`.

The long forms always have the strings in the following order
```latex
\lbx@lfromlang išvertė
```
But for the short form, some string have
```latex
\lbx@sfromlang vert\adddot
```
and some have
```latex
vert\adddotspace \lbx@sfromlang
```
or
```latex
vert\adddot\ \lbx@sfromlang
```
`\lbx@lfromlang`/ `\lbx@sfromlang` either expand to nothing, if the `origlanguage` field is not present or to "`from<origlanguage>`" if it is.

Compare
```latex
\documentclass[lithuanian]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=authoryear, backend=biber, abbreviate=true]{biblatex}


\begin{filecontents}[force]{\jobname.bib}
@book{appleby,
  author       = {Humphrey Appleby},
  title        = {On the Importance of the Civil Service},
  titleaddon   = {With \texttt{origlanguage}},
  date         = {1980},
  translator   = {Bernard Woolley},
  origlanguage = {latin},
}
@book{bppleby,
  author       = {Humphrey Bppleby},
  title        = {On the Importance of the Civil Service},
  titleaddon   = {Without \texttt{origlanguage}},
  date         = {1980},
  translator   = {Bernard Woolley},
}
@book{cppleby,
  author       = {Humphrey Cppleby},
  title        = {On the Importance of the Civil Service},
  titleaddon   = {With \texttt{origlanguage}},
  date         = {1980},
  editor       = {Bernard Woolley},
  translator   = {Bernard Woolley},
  origlanguage = {latin},
}
@book{dppleby,
  author       = {Humphrey Dppleby},
  title        = {On the Importance of the Civil Service},
  titleaddon   = {Without \texttt{origlanguage}},
  date         = {1980},
  editor       = {Bernard Woolley},
  translator   = {Bernard Woolley},
}
\end{filecontents}
\addbibresource{\jobname.bib}

\begin{document}
\nocite{*}
\printbibliography
\end{document}
```
with `abbreviate=false,` and `abbreviate=true,`. In particular the order of "iš lot. k." and "vert." in the entries `appleby` and `cppleby`.

**Comment by maras** [comment]
> I take it the "m." and "d." that appear in the dates are the Lithuanian words for year and day? Can you confirm that? 
Yes.
> Do we need to differentiate (in theory) between "year"/"day" in nominative or "year"/"day" in genitive?
In theory yes, but only in full forms (nom. – metai/diena, gen. – metų/dienos), in practice I know only where metai/metų can be used.

> I can try to look into [year] m. [month genitivus] mėn. if that is "more correct" than "[year] m. [month nominativus]".
It depends. Date with gen. can be used for archival documents, but for newspapers or magazines better to use with nom. Ideal solution would be two forms of full dates when only year and month are – with nom and with gen. So we need both nom. and gen.
 And one more nitpick is that in [year] m. [month genitivus] mėn. case "mėn" should be only after last month name. So not "2019 m. liep. mėn. –- rugs. mėn." but "2019 m. liep. –- rugs. mėn."
Is that too not too cumbersome? 

> 1. February--6. March 2010 would be 
2010 m. vasario 1 d. -- kovo 6 d. (endash).

And your examples are all good except last, as I wrote.

> if there is no equivalent to "a.m." and "p.m." in Lithuanian it is better to leave the strings untranslated and commented out.
OK.

> By default biblatex doesn't use ordinals in a lot of places anyway. They are basically only used for editions (\mkbibordedition), series (\mkbibordseries) 
In Lithuanian too, I think. And that's very good.

> How would you say
>    5th edition
5-as leid. (or 5-as leidimas)
>    Volume 6
In bibliographies: mostly t. 6 or (rarely) tomas 6. In regular text: 6-as tomas or šeštas tomas. 
>    Chapter 3
3 skyrius, Trečias skyrius. I don't remember case with ordinals, it would be 3-as skyrius
>    pp. 245-260
p. 245-260 (very very rarely puslapiai 245-260)
>    Version 2
I don't know for shure, but: vers. 2 or 2-a vers. or 2-a versija (version – fem.)

For csquotes I am using german style now, but I'll submit.

> Is the difference between short and long form "U.S." vs "JAV" in 
  patentus         = {{U.S\adddotspace patentas}{JAV\addspace pat\adddot}}, and   patrequs         = {{U.S\adddotspace patento prašymas}{JAV\addspace pat\adddot\ praš\adddot}}, intentional?
No, that's my mistake. I don't know anything about patents, but it should be patentus         = {{JAV\addspace patentas}{JAV\addspace pat\adddot}}, and   patrequs         = {{JAV\addspace patento prašymas}{JAV\addspace pat\adddot\ praš\adddot}}

Inconsistencies with "vert." and "išvertė" are intentional: full form would be: "iš lotynų kalbos išvertė", short: "vert. iš lot. k." or "iš lot. k. vert." – I need to search more examples, because I don't know where to put "vert." (it can be placed in both places in Lithuanian – my language is quite free about word order in the sentence).

Thank you for your work.

**Comment by moewew** [comment]
Thanks for your reply.

> > Do we need to differentiate (in theory) between "year"/"day" in nominative or "year"/"day" in genitive?
> 
> In theory yes, but only in full forms (nom. – metai/diena, gen. – metų/dienos), in practice I know only where metai/metų can be used.

OK. Do we need the long forms for the dates, though? All date examples so far only have "m." and "d." and possibly "mėn.". Would it ever be natural to give the full word in the date?


> > I can try to look into [year] m. [month genitivus] mėn. if that is "more correct" than "[year] m. [month nominativus]".
> 
> It depends. Date with gen. can be used for archival documents, but for newspapers or magazines better to use with nom. Ideal solution would be two forms of full dates when only year and month are – with nom and with gen. So we need both nom. and gen.

But if we have two versions of long `yyyy-mm` dates one with nominative and one with genitive, how should `biblatex` know which date version to take?


> And one more nitpick is that in [year] m. [month genitivus] mėn. case "mėn" should be only after last month name. So not "2019 m. liep. mėn. –- rugs. mėn." but "2019 m. liep. –- rugs. mėn."
> Is that too not too cumbersome?

Done!


> 
> > 1. February--6. March 2010
> 
> would be 2010 m. vasario 1 d. -- kovo 6 d. (endash).
> And your examples are all good except last, as I wrote.

Good!


> > Is the difference between short and long form "U.S." vs "JAV" in
> > patentus         = {{U.S\adddotspace patentas}{JAV\addspace pat\adddot}}, and   patrequs         = {{U.S\adddotspace patento prašymas}{JAV\addspace pat\adddot\ praš\adddot}}, intentional?
> 
> No, that's my mistake. I don't know anything about patents, but it should be patentus = {{JAV\addspace patentas}{JAV\addspace pat\adddot}}, and patrequs = {{JAV\addspace patento prašymas}{JAV\addspace pat\adddot\ praš\adddot}}

Thanks for the confirmation.


> Inconsistencies with "vert." and "išvertė" are intentional: full form would be: "iš lotynų kalbos išvertė", short: "vert. iš lot. k." or "iš lot. k. vert." – I need to search more examples, because I don't know where to put "vert." (it can be placed in both places in Lithuanian – my language is quite free about word order in the sentence).

I see. But is it really wise to have the inconsistencies in
```latex
\documentclass[lithuanian]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=authoryear, backend=biber, abbreviate=true]{biblatex}


\begin{filecontents}[force]{\jobname.bib}
@book{appleby,
  author       = {Humphrey Appleby},
  title        = {On the Importance of the Civil Service},
  titleaddon   = {With \texttt{origlanguage}},
  date         = {1980},
  translator   = {Bernard Woolley},
  origlanguage = {latin},
}
@book{cppleby,
  author       = {Humphrey Cppleby},
  title        = {On the Importance of the Civil Service},
  titleaddon   = {With \texttt{origlanguage}},
  date         = {1980},
  editor       = {Bernard Woolley},
  translator   = {Bernard Woolley},
  origlanguage = {latin},
}
\end{filecontents}
\addbibresource{\jobname.bib}

\begin{document}
\nocite{*}
\printbibliography
\end{document}
```
> Iš lot. k. vert. Bernard Woolley.
> Sud. ir vert. iš lot. k. Bernard Woolley.

It would be great to decide on a place for the "vert." in all strings. So either

> Iš lot. k. vert. Bernard Woolley.
> Sud. ir iš lot. k. vert. Bernard Woolley.

or

> Vert. iš lot. k. Bernard Woolley.
> Sud. ir vert. iš lot. k. Bernard Woolley.

I'm asking also because the macros `\lbx@lfromlang` and `\lbx@sfromlang` are a bit special and need to take care of some spacing. So this inconsistency can be problematic for coding.

---

Once I have worked through your replies I will probably open a pull request with my version of `lithuanian.lbx` that you can then inspect, test and comment on and if you are happy with it, we can merge it.

Because I worked with a copy of your file from https://github.com/maras/lithuanian.lbx/blob/master/lithuanian.lbx and not with a branch of the `biblatex` repository the system will probably not show you as the author of the commit. I hope that is OK for you. If you would like to be the commit author, we will can come up with a workaround.

**Comment by maras** [comment]
Sorry for delay – I was offline. I'll be reachable more rapidly till now.

> Do we need the long forms for the dates, though? All date examples so far only have "m." and "d." and possibly "mėn.". Would it ever be natural to give the full word in the date?

I don't think, that we need – all bibliography style examples are only with short forms m. and d. in the long form of dates.

> But if we have two versions of long yyyy-mm dates one with nominative and one with genitive, how should biblatex know which date version to take?

I checked once more and some bibliographies are using `month name nom.` and other `[month name gen.] mėn.` in `yyyy-mm` dates. I can't find any consistency and that drives me crazy. BUT there is very rare examples of `2020-03`, more often – `2020 kovas` in short date form. So maybe we can render `yyyy-mm` dates in such way: `[year] m. [month genitivus] mėn.` for long form and `[year] m. [month nominativus]` for short form? User will have possibility to pick up correct date format for his needs.

> It would be great to decide on a place for the "vert." in all strings.

I understand and consistency is always better. So I am for
> Iš lot. k. vert. Bernard Woolley.
> Sud. ir iš lot. k. vert. Bernard Woolley.

version.

> Once I have worked through your replies I will probably open a pull request with my version of lithuanian.lbx that you can then inspect, test and comment on and if you are happy with it, we can merge it.

OK.

> Because I worked with a copy of your file from https://github.com/maras/lithuanian.lbx/blob/master/lithuanian.lbx and not with a branch of the biblatex repository the system will probably not show you as the author of the commit. I hope that is OK for you.

Shure, I don't need a fame :).

Thank your for such attention to details.

**Comment by moewew** [comment]
> > Do we need the long forms for the dates, though? All date examples so far only have "m." and "d." and possibly "mėn.". Would it ever be natural to give the full word in the date?
> 
> I don't think, that we need – all bibliography style examples are only with short forms m. and d. in the long form of dates.

OK, we'll stick with one form (`year`, `month`, `day`) and translate it with the abbreviation in both short and long form.

> > But if we have two versions of long yyyy-mm dates one with nominative and one with genitive, how should biblatex know which date version to take?
> 
> I checked once more and some bibliographies are using `month name nom.` and other `[month name gen.] mėn.` in `yyyy-mm` dates. I can't find any consistency and that drives me crazy. BUT there is very rare examples of `2020-03`, more often – `2020 m. kovas` in short date form. So maybe we can render `yyyy-mm` dates in such way: `[year] m. [month genitivus] mėn.` for long form and `[year] m. [month nominativus]` for short form? User will have possibility to pick up correct date format for his needs.

Mhhh, yes that would be a possibility, but I would find it odd that the short form would say

> 2020

> 2020 m. kovas

> 2020-03-18

for `2020`, `2020-03` and `2020-03-18`, respectively. It just doesn't look particularly consistent to me.

But if one really never says

> 2020-03

then that might be an option.

> 
> > It would be great to decide on a place for the "vert." in all strings.
> 
> I understand and consistency is always better. So I am for
> 
> > Iš lot. k. vert. Bernard Woolley.
> > Sud. ir iš lot. k. vert. Bernard Woolley.
> 
> version.

Very good. Done!


> > Because I worked with a copy of your file from https://github.com/maras/lithuanian.lbx/blob/master/lithuanian.lbx and not with a branch of the biblatex repository the system will probably not show you as the author of the commit. I hope that is OK for you.
> 
> Shure, I don't need a fame :).

Phew. :-) That saves us a bit of back and forth.

**Comment by maras** [comment]
> OK, we'll stick with one form (`year`, `month`, `day`) and translate it with the abbreviation in both short and long form.

There is always a possibility to create own `.lbx` file with custom forms of needed abbreviations.

> It just doesn't look particularly consistent to me.

For me too. And some authors says `2020-03` and `2020-03/04` to distinguish `2020 march – may` from `2020-03-04` (`4th of march 2020`). Maybe that can be the way? That would be short form of date and `[year] m. [month genitivus] mėn.` – long.

**Comment by moewew** [comment]
> > It just doesn't look particularly consistent to me.
> 
> For me too. And some authors says `2020-03` and `2020-03/04` to distinguish `2020 march – may` from `2020-03-04` (`4th of march 2020`). Maybe that can be the way? That would be short form of date and `[year] m. [month genitivus] mėn.` – long.

I guess

> 2020-03/04

would look OK to me, but

> 2020-03/05

looks a bit odd (if one is not used to ISO 8601 using `/` as the range separator).

But I admit that currently the difference between `2020-03-05` and `2020-03/2020-05` is only

> 2020-03-05

> 2020-03--05

and can be quickly overlooked.

**Comment by maras** [comment]
Yes, it is quite strange to see 
> 2020-03/05

I can ask Commision of Lithuanian Language (Lithuania have such institution) how to write such dates. [There](http://www.vlkk.lt/aktualiausios-temos/rasyba/data-ir-laikas) is Commisions clarification how to write dates, but nothing about what to do with short dates when only year and month is known. Those clarifications are general, not for bibliographies.

**Comment by moewew** [comment]
We just added support for Turkish in https://github.com/plk/biblatex/pull/975. Could you tell me the translations for
* `langturkish`
* `fromturkish`
please?

**Comment by maras** [comment]
`langturkish`
turkų kalba/k.

    `fromturkish`
iš turkų kalbos/k.
There is no short form – too short word.

**Comment by moewew** [comment]
Thanks, so I guess we only need to decide what to do with short dates of the form `YYYY-MM`?

**Comment by maras** [comment]
Yes, `YYYY-MM` is the problem. I wrote to the Commision, but no answer. There are three variants, I think:
> [year] m. [month nominativus]

It is most inconsistent, bu it is used quite wide (maybe most widely).
> YYYY-MM--MM

This hard to distinguish from `YYYY-MM-DD` while it is consistent with other short date ranges.
> YYYY-MM/MM

This is quite unusual, but comply with international ISO date standard which is used in Lithuania. And using it means that other short date ranges should be with / too `YYYY-MM-DD/MM-DD`, `YYYY-MM-DD/DD`.

What format is prevaling in biblatex short date ranges?

**Comment by moewew** [comment]
> > [year] m. [month nominativus]
> 
> It is most inconsistent, bu it is used quite wide (maybe most widely).

Well, if it's the most widely used, then we may have to take it. It is unfortunate that things will look inconsistent (not only because it looks odd, but also because it is a bit of a pain to implement), but what can we do?

> 
> > YYYY-MM--MM
> 
> This hard to distinguish from `YYYY-MM-DD` while it is consistent with other short date ranges.
> 
> > YYYY-MM/MM
> 
> This is quite unusual, but comply with international ISO date standard which is (or should be) used in Lithuania. And using it means that other short date ranges should be with / too `YYYY-MM-DD/MM-DD`, `YYYY-MM-DD/DD`.

I see ISO 8601 as a standard for machine-readable dates rather than a standard for human-friendly date formats. So if you want to go full-on ISO 8601 with "/" as range separator that is certainly doable, but if it is not commonly done in Lithuanian, I'm not sure if I would do it.

> What format is prevaling in biblatex short date ranges?

That's not the question we should ask here. The question is what format is prevailing *in Lithuanian*.

Plus it's not a question that can be answered properly, since the short date format depends on the language (German `YYYY-MM` short dates would be "03/2020", but `YYYY-MM-DD` would be "18.03.2020", which looks OK to me; British and American `YYYY-MM` dates are "03/2020", but I'm not sure if that is legit; obviously those formats are woefully inadequate for Lithuanian).

**Comment by maras** [comment]
> That's not the question we should ask here. The question is what format is prevailing _in Lithuanian_.

I understand, but the trickiest thing is that there is no prevailing format: even the same publisher (academic) is using different styles for different books. It depends on redactors or authors (mostly) will. It's a mess. I'm a historian and all Lithuanian historical and humanitarian journals had different bibliographic rules. These differences are small mostly but they exist. There are some common tendencies that I tried to reflect.

To overcome similarity of `YYYY-MM--MM` to `YYYY-MM-DD` we can remove hyphens between date parts completely for all short dates. Endash without spaces would reflect  date ranges. Some styles use hyphens in short dates, some not, it is hard to say which style is prevailing. Maybe that could be the way?

**Comment by moewew** [comment]
OK, spaces are an easy way out. In the end a user can still change things if they need.

You can download the current test version of `lithuanian.lbx` at https://gist.github.com/moewew/ceeb0c2beff7af149ebc53ee15e379cf.

Since that version defines new strings that are not in `biblatex` v3.14, you'll have to temporarily add the following lines to your `.tex` document for testing.
```latex
\NewBibliographyString{
  langlithuanian,
  langserbian,
  langturkish,
  fromlithuanian,
  fromserbian,
  fromturkish,
}
```

Naturally this won't be necessary in the release version that includes `lithuanian.lbx`.

Dates can be tested in
```latex
\documentclass[lithuanian]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=authoryear, backend=biber, alldates=comp, labeldate=year]{biblatex}

\NewBibliographyString{
  langlithuanian,
  langserbian,
  langturkish,
  fromlithuanian,
  fromserbian,
  fromturkish,
}

\begin{filecontents}{\jobname.bib}
@book{appleby,
  author  = {Humphrey Appleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-02-01/2020-02-06},
}
@book{bppleby,
  author  = {Humphrey Bppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-02-01/2020-03-06},
}
@book{cppleby,
  author  = {Humphrey Cppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2019-03-05/2020-01-07},
}
@book{dppleby,
  author  = {Humphrey Dppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2019-07/2019-09},
}
@book{eppleby,
  author  = {Humphrey Eppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020},
}
@book{fppleby,
  author  = {Humphrey Fppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-03},
}
@book{gppleby,
  author  = {Humphrey Gppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-03-18},
}
@book{hppleby,
  author  = {Humphrey Hppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-03/2020-05},
}
@book{ippleby,
  author  = {Humphrey Ippleby},
  title   = {On the Importance of the Civil Service},
  date    = {2019/2020},
}
\end{filecontents}
\addbibresource{\jobname.bib}


\begin{document}
\cite{appleby,bppleby,cppleby,dppleby}

\cite{eppleby,fppleby,gppleby,hppleby,ippleby}

\printbibliography
\end{document}
```
Change the `alldates` option values to `terse`, `long`, `short` to see different outcomes.

**Comment by maras** [comment]
Outcomes of comp, terse and short options are good, but endash should be with spaces in long option (as they are in comp option now).
How can I test long form of month names, because now I see only shortened versions of month in comp and long options?

**Comment by moewew** [comment]
> Outcomes of comp, terse and short options are good, but endash should be with spaces in long option (as they are in comp option now).

Oh, thank you very much for catching that, I had totally forgotten about that.

Please try the updated `lithuanian.lbx` from https://gist.github.com/moewew/ceeb0c2beff7af149ebc53ee15e379cf

> How can I test long form of month names, because now I see only shortened versions of month in comp and long options?

The option you are looking for is called `dateabbrev=false,`
```latex
\documentclass[lithuanian]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=authoryear, backend=biber, alldates=long, dateabbrev=false, labeldate=year]{biblatex}

\NewBibliographyString{
  langlithuanian,
  langserbian,
  langturkish,
  fromlithuanian,
  fromserbian,
  fromturkish,
}

\begin{filecontents}{\jobname.bib}
@book{appleby,
  author  = {Humphrey Appleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-02-01/2020-02-06},
}
@book{bppleby,
  author  = {Humphrey Bppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-02-01/2020-03-06},
}
@book{cppleby,
  author  = {Humphrey Cppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2019-03-05/2020-01-07},
}
@book{dppleby,
  author  = {Humphrey Dppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2019-07/2019-09},
}
@book{eppleby,
  author  = {Humphrey Eppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020},
}
@book{fppleby,
  author  = {Humphrey Fppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-03},
}
@book{gppleby,
  author  = {Humphrey Gppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-03-18},
}
@book{hppleby,
  author  = {Humphrey Hppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-03/2020-05},
}
@book{ippleby,
  author  = {Humphrey Ippleby},
  title   = {On the Importance of the Civil Service},
  date    = {2019/2020},
}
\end{filecontents}
\addbibresource{\jobname.bib}


\begin{document}
\cite{appleby,bppleby,cppleby,dppleby}

\cite{eppleby,fppleby,gppleby,hppleby,ippleby}

\printbibliography
\end{document}
```

**Comment by maras** [comment]
It is all right with terse and short options.
Long and comp options:
> Appleby Humphrey (2020 m. vasario 1 d. – 2020 m. vasario 6 d.)

should be
> Appleby Humphrey (2020 m. vasario 1–6 d.)

and
> Hppleby Humphrey (2020 m. kovo mėn. – 2020 m. gegužės mėn.)

shoud be
> Hppleby Humphrey (2020 m. kovo – gegužės mėn.)

> Ippleby Humphrey (2019 m. – 2020 m.).

should be 
> Ippleby Humphrey (2019–2020 m.).

Thank you for your work!

**Comment by moewew** [comment]
> > Ippleby Humphrey (2019 m. – 2020 m.).
> 
> should be
> 
> > Ippleby Humphrey (2019–2020 m.).

Done!

---

I'm a bit weary about the other changes. If the should apply to both `long` and `comp`, then what is the difference between `long` and `comp`?

**Comment by maras** [comment]
> I'm a bit wary about the other changes. If they should apply to both long and comp, then what is the difference between long and comp?

I'm using only verbose styles, so my knowledge is quite limited, but as I understand, the only difference between them is long or abbreviated month names, but that doesn't affect position of endash – it should be the same: when there is a word (abbreviated with a dot too) from one side, there should be endash with spaces, when there are cyphers from both side – endash without spaces. So comp dates should be
> Appleby Humphrey (2020 m. vas. 1–6 d.)
> Hppleby Humphrey (2020 m. kovo – geg. mėn.)

**Comment by moewew** [comment]
No `long` and `comp` are not about abbreviation of month names, that's what `dateabbrev` is for. `long` means that `biblatex` makes no attempt to make a range more compact, the output will always be in the fullest form. `comp` on the other hand, tries to make the date more compact by leaving out certain parts of the date. You can see the intended difference clearly when you run the example with `ngerman` and compare `alldates=long,` to `alldates=comp,`
```latex
\documentclass[ngerman]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=authoryear, backend=biber, alldates=comp, dateabbrev=false, labeldate=year]{biblatex}

\NewBibliographyString{
  langlithuanian,
  langserbian,
  langturkish,
  fromlithuanian,
  fromserbian,
  fromturkish,
}

\begin{filecontents}{\jobname.bib}
@book{appleby,
  author  = {Humphrey Appleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-02-01/2020-02-06},
}
@book{bppleby,
  author  = {Humphrey Bppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-02-01/2020-03-06},
}
@book{cppleby,
  author  = {Humphrey Cppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2019-03-05/2020-01-07},
}
@book{dppleby,
  author  = {Humphrey Dppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2019-07/2019-09},
}
@book{eppleby,
  author  = {Humphrey Eppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020},
}
@book{fppleby,
  author  = {Humphrey Fppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-03},
}
@book{gppleby,
  author  = {Humphrey Gppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-03-18},
}
@book{hppleby,
  author  = {Humphrey Hppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-03/2020-05},
}
@book{ippleby,
  author  = {Humphrey Ippleby},
  title   = {On the Importance of the Civil Service},
  date    = {2019/2020},
}
\end{filecontents}
\addbibresource{\jobname.bib}


\begin{document}
\cite{appleby,bppleby,cppleby,dppleby}

\cite{eppleby,fppleby,gppleby,hppleby,ippleby}

\printbibliography
\end{document}
```

As I understand it what we are currently gravitating towards is a setup where `long` and `comp` are pretty much the same.

**Comment by maras** [comment]
I see now, didn't know that (and I can't find anything about that in biblatex manual). Most styles are using comp, but some are using long. We can create different long and comp rules to encompass the more varieties we can, but I don't dare bother you more :).

**Comment by moewew** [comment]
The `date` option is documented on pp. 54-55 of the manual (in v3.14).

Since `biblatex` offers both `long` and `comp` it would be better to implement both properly. (Even if none of the standard styles uses `long` by default, almost all dates use the `comp` form by default, `urldate` is `short` and `labeldate` is `year`.) So if you are still in for a bit of back and forth I'd like to implement `long`. (The cheap option would be to just make `long` do the same as `comp`, but I would only consider that a viable option if the overwhelming majority of Lithuanian style uses `comp` and there is no real notion of `long` date ranges.)

**Comment by maras** [comment]
Somehow my Ocular can't find " comp " in that pages, strange.

OK, let's try.
Long:
> Appleby Humphrey (2020 m. vasario 1 d. – 2020 m. vasario 6 d.)

comp:
> Appleby Humphrey (2020 m. vasario 1–6 d.)

long:
> Bppleby Humphrey (2020 m. vasario 1 d. – 2020 m. kovo 6 d.). 

comp:
> Bppleby Humphrey (2020 m. vasario 1 d. – kovo 6 d.). 

Long:
> Hppleby Humphrey (2020 m. kovo mėn. – 2020 m. gegužės mėn.)

comp: 
> Hppleby Humphrey (2020 m. kovo – gegužės mėn.)

and long
> Ippleby Humphrey (2019 m. – 2020 m.).

comp:
> Ippleby Humphrey (2019–2020 m.).

I think, that's all.

**Comment by moewew** [comment]
Ah that doesn't look so bad. Please have another try with https://gist.github.com/moewew/ceeb0c2beff7af149ebc53ee15e379cf.

**Comment by maras** [comment]
I think dates are OK now.

I compiled localisation-keys once more and found some typos:
> 195 {sud\adddotspace ir prat\adddotspace auto\adddot}}, 

should be
> > 195 {sud\adddotspace ir prat\adddotspace aut\adddot}}, 

and
> 331   withintroduction = {{įvado autorius}{įvad\adddotspace aut\adddot}},

should be
>  331  withintroduction = {{įvado autorius}{įv\adddotspace aut\adddot}},

I sent this file to other humanitarian, maybe she  will see some typos or mistakes.

**Comment by moewew** [comment]
Very good.

https://gist.github.com/moewew/ceeb0c2beff7af149ebc53ee15e379cf is updated with the two string changes.

Let me know when this is ready to merge or if something else needs to  be done.

**Comment by maras** [comment]
Two more strings:
>   585 countryeu        = {{Europos sąjunga}{EU}},

should be
>   585 countryeu        = {{Europos Sąjunga}{EU}},

and
> 586 countryep        = {{Europos sąjunga}{EP}},

should be
> 586 countryep        = {{Europos Sąjunga}{EP}},

And, I think, it could be merged. At last! :)

And one more question: how can I change month name from gen. to nom. when date is `yyyy-mm`
> Hppleby Humphrey (2020 m. kovo mėn. – 2020 m. gegužės mėn.)

to 
> Hppleby Humphrey (2020 m. kovas – 2020 m. gegužė)

and so on in the style file or in custom `.lbx`? This can be quite useful in some styles.

**Comment by moewew** [comment]
Last corrections applied and some reworking done in https://gist.github.com/moewew/ceeb0c2beff7af149ebc53ee15e379cf.

With that `.lbx` I think the following redefinition of `\mkbibdatelong` should give nominative months
```latex
\documentclass[lithuanian]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=authoryear, backend=biber, alldates=comp, dateabbrev=false, labeldate=year]{biblatex}

\makeatletter
\DefineBibliographyExtras{lithuanian}{%
  \protected\def\mkbibdatelong#1#2#3{%
    \iffieldundef{#1}
      {}
      {\iffieldbibstring{#1}
         {\bibstring{\thefield{#1}}}
         {\dateeraprintpre{#1}%
          \stripzeros{\thefield{#1}}%
          \lbx@lt@ifprintyearstring
            {}
            {\addnbspace\bibstring{year}}}%
       \iffieldundef{#2}{}{\addnbspace}}%
    \iffieldundef{#2}
      {}
      {\iffieldundef{#3}
         {\mkbibmonth{\thefield{#2}}}
         {\lbx@lt@mkgenmonth{\thefield{#2}}%
          \addnbspace}}%
    \iffieldundef{#3}
      {}
      {\stripzeros{\thefield{#3}}%
       \lbx@lt@ifprintdaystring{}{\addnbspace\bibstring{day}}}%
  }%
}
\makeatother

\NewBibliographyString{
  langlithuanian,
  langserbian,
  langturkish,
  fromlithuanian,
  fromserbian,
  fromturkish,
}

\begin{filecontents}{\jobname.bib}
@book{appleby,
  author  = {Humphrey Appleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-02-01/2020-02-06},
}
@book{bppleby,
  author  = {Humphrey Bppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-02-01/2020-03-06},
}
@book{cppleby,
  author  = {Humphrey Cppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2019-03-05/2020-01-07},
}
@book{dppleby,
  author  = {Humphrey Dppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2019-07/2019-09},
}
@book{eppleby,
  author  = {Humphrey Eppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020},
}
@book{fppleby,
  author  = {Humphrey Fppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-03},
}
@book{gppleby,
  author  = {Humphrey Gppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-03-18},
}
@book{hppleby,
  author  = {Humphrey Hppleby},
  title   = {On the Importance of the Civil Service},
  date    = {2020-03/2020-05},
}
@book{ippleby,
  author  = {Humphrey Ippleby},
  title   = {On the Importance of the Civil Service},
  date    = {2019/2020},
}
\end{filecontents}
\addbibresource{\jobname.bib}


\begin{document}
\cite{appleby,bppleby,cppleby,dppleby}

\cite{eppleby,fppleby,gppleby,hppleby,ippleby}

\printbibliography
\end{document}
```

**Comment by maras** [comment]
Yes, all is perfectly working.
Thank you very much for your work and patience!

**Comment by moewew** [comment]
Thank you for the Lithuanian localisation.

I opened a pull request for the `.lbx` file and some additional stuff at https://github.com/plk/biblatex/pull/977. If you are happy with what you see there, we can merge.

**Comment by maras** [comment]
I think it's OK. Thank you once more :).

**Comment by moewew** [comment]
One (hopefully) last thing, I just noted that in

https://github.com/plk/biblatex/blob/66e7dede971a28e781e55b70bbd0a98cf2f33be2/tex/latex/biblatex/lbx/lithuanian.lbx#L479

the short form
```latex
nuo šiol –\addspace
```
ends with a space command. This is usually discouraged: The only trailing punctuation in bibstrings should be `\adddot`, everything else should be taken care of when the string is used. In particular you can expect normal uses of `\bibstring{citedas}` to be followed by either a space or a colon and a space anyway. In both cases the `\addspace` will look weird.

I also find the dash at the end a bit unusual, it can cause problems when the style decides to follow `\bibstring{citedas}` by a colon and a space and not just a space. In light of this it may be nicer to find a string that does not end in `–\addspace`.

Example document: Note that with the current definition `nuo šiol –\addspace` there are two spaces before the `shorthand` in the citation
```latex
\documentclass[lithuanian]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=verbose, backend=biber]{biblatex}

\NewBibliographyString{
  langlithuanian,
  langserbian,
  langturkish,
  fromlithuanian,
  fromserbian,
  fromturkish,
}

\DefineBibliographyStrings{lithuanian}{
  citedas = {nuo šiol –\addspace},
}

\addbibresource{biblatex-examples.bib}

\begin{document}
\cite{kant:kpv}
\printbibliography
\end{document}
```

**Comment by maras** [comment]
I see. We can change 
>  citedas          = {{nuo šiol cituojamas kaip}{nuo šiol –\addspace}}, 

to
> citedas          = {{nuo šiol cituojamas kaip}{nuo šiol}},

and let punctuation to styles.

Thank you!

**Comment by moewew** [comment]
Thanks. #977 is merged now. `lithuanian.lbx` should be in the next release version of `biblatex`.

**Comment by maras** [comment]
Thank you!

**Comment by moewew** [comment]
`biblatex` 3.15 has been released and is available in MikTeX and TeX live now. Lithuanian should now work out of the box.

Thank you very much for your contribution. In case anything is missing, just open a new issue.
